contributing/core/testing.md

```diff
@@ -0,0 +1,70 @@
+# Testing
+
+## Running tests
+
+We recommend running the tests in headless mode (with the browser windows hidden) and with a specific directory pattern (`--testPathPattern`) or test name (`-t`) which ensures only a small part of the test suite is run locally:
+
+For example, running one test in the production test suite:
+
+````
+Running one test in the `test/integration/production` test suite:
+
+```sh
+pnpm test --testPathPattern "test/integration/production" -t "should allow etag header support"
+````
+
+Running all tests in the `test/integration/production` test suite:
+
+```sh
+pnpm test --testPathPattern "test/integration/production"
+```
+
+Running all tests (⚠️ not recommended locally):
+
+```sh
+pnpm test
+```
+
+When you want to debug a particular test you can replace `pnpm test` with `pnpm testonly` to opt out of the headless browser.
+When the test runs it will open the browser that is in the background by default, allowing you to inspect what is on the screen.
+
+```sh
+pnpm testonly --testPathPattern "test/integration/production" -t "should allow etag header support"
+```
+
+## Writing tests for Next.js
+
+### Getting Started
+
+You can set up a new test using `pnpnm new-test` which will start from a template related to the test type.
+
+### Test Types in Next.js
+
+- e2e: These tests will run against `next dev`, `next start`, and deployed to Vercel
+- development: These tests only run against `next dev`
+- production: These tests will run against `next start`.
+- integration: These tests run misc checks and modes and it is where tests were added historically. Ideally, we don't add new test suites here anymore as these tests are not isolated from the monorepo.
```

Comment:
This conflicts with the pull request template for bugs ([here])(https://github.com/vercel/next.js/blob/46dd1cffcfc87a07d40ec3d0436ab57a7e22def3/.github/pull_request_template.md?plain=1#L10), does it need updating?